cri-o-testonly.spec

```diff
@@ -0,0 +1,273 @@
+# This file is only used for running automated rpm build tests for PRs
+# The canonical dist-git source can be found at:
+# https://src.fedoraproject.org/rpms/cri-o/blob/master/f/cri-o.spec
+
+%if 0%{?with_debug}
+%global _dwz_low_mem_die_limit 0
+%else
+%global debug_package   %{nil}
+%endif
+
+%global provider        github
+%global provider_tld    com
+%global project         kubernetes-incubator
+%global repo            cri-o
+# https://github.com/kubernetes-incubator/cri-o
+%global provider_prefix %{provider}.%{provider_tld}/%{project}/%{repo}
+%global import_path     %{provider_prefix}
+%global commit0         bb1da978e30b3d32f988e30568b5161e8cfd700b
+%global shortcommit0    %(c=%{commit0}; echo ${c:0:7})
+
+%global service_name crio
+
+Name:           cri-o-testonly
+Version:        1.0.0
+Release:        2.testonly%{?dist}
+Summary:        OCI-based implementation of Kubernetes Container Runtime Interface
+License:        ASL 2.0
+URL:            https://%{provider_prefix}
+Source0:        %{name}.tar.gz
+ExcludeArch:    ppc64
+# If go_compiler is not set to 1, there is no virtual provide. Use golang instead.
+BuildRequires:  %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires:  btrfs-progs-devel
+BuildRequires:  git
+BuildRequires:  glib2-devel
+BuildRequires:  glibc-static
+BuildRequires:  go-md2man
+BuildRequires:  gpgme-devel
+BuildRequires:  libassuan-devel
+BuildRequires:  libseccomp-static
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  device-mapper-devel
+BuildRequires:  make
+Requires(pre):  container-selinux
+Requires:       skopeo-containers
+Requires:       runc >= 2:1.0.0-10
+Provides:       %{service_name} = %{version}-%{release}
+Obsoletes:      ocid <= 0.3
+Provides:       ocid = %{version}-%{release}
+Recommends: containernetworking-plugins >= 0.5.2-2
+
+%description
+%{summary}
+
+%prep
+%autosetup -Sgit -n %{name}
```

Comment:
How does this affect the logic in the top-level Makefile that sets the `COMMIT_NO` and `GIT_COMMIT` variables?

Reply:
the `git archive` step in the rpm target should just create a tarball of the HEAD commit and that's the source tarball used here. I haven't used `COMMIT_NO` or `GIT_COMMIT` variables. Do I need to?

Reply:
IIUC, using `%autosetup -Sgit` generates a repository using the exploded tarball, usin git as an initial commit, and applying any patches listed in the .spec file as additional commits.  The resulting HEAD commit ID, which `crio --version` displays, would be unlikely to match a revision in the repository on github.

Reply:
I'll switch to using `%setup`. This was just something copied over from the canonical spec from fedora.